libs/ledger/benchmark/executor_bench.cpp

```diff
@@ -80,9 +80,9 @@ void Executor_BasicBenchmark(benchmark::State &state)
 
     TokenContract tokens{};
 
-    tokens.Attach({&tokens, tx->contract_address(), &adapter, 0});
+    fetch::ledger::ContractContext      context{&tokens, tx->contract_address(), &adapter, 0};
+    fetch::ledger::ContractAttachHelper raii(tokens, context);
```

Comment:
One of minor issues I see in this implementation is that context attachment is done in two statements in most cases. Even though one of them is simply an object instantiation, still RAII is usually perceived as something more compact, one statement to lock context until the end of the block.
One possible alternative would be something like
```cpp
template<class... Args> ContractAttachHelper(Contract &contract, Args &&...args)
    : contract_(contract)
{
    contract_.Attach(ContractContext(std::forward<Args>(args)...));
}
```
Next, contexts here in most use-cases are often short-living locals. `Contract::Attach()` could attach to them by-value rather than copying each time.

Reply:
I switched to passing by value; I disagree on the template constructor - it needlessly increases coupling between the `Context` and the `Attacher`. I would rather keep the `Attacher` class simple, along with the stepwise constructing and then attaching a context. To do otherwise seems to sacrifice clarity for brevity.